internal/state/graph/httproute.go

```diff
@@ -405,20 +406,31 @@ func findAcceptedHostnames(listenerHostname *v1beta1.Hostname, routeHostnames []
 		if hostname == "" {
 			return true
 		}
-		return string(h) == hostname
+
+		routeHost := string(h)
+		return routeHost == hostname || wildcardMatch(hostname, routeHost) || wildcardMatch(routeHost, hostname)
```

Comment:
Can we simplify this a bit and make it easier to use? I think calling `wildcardMatch` twice is a little confusing. Maybe we can pull the `match` function out of `findAcceptedHostnames` and combine it with `wildcardMatch`?

Reply:
Something like this?

```
func match(listenerHost, routeHost string) bool {
	if listenerHost == "" {
		return true
	}

	if routeHost == listenerHost {
		return true
	}

	if strings.HasPrefix(listenerHost, "*.") && strings.HasSuffix(routeHost, strings.TrimPrefix(listenerHost, "*")) {
		return true
	}

	return strings.HasPrefix(routeHost, "*.") && strings.HasSuffix(listenerHost, strings.TrimPrefix(routeHost, "*")) {
}
```

Reply:
Either way sounds good to me!

Reply:
I'll still be calling wildcardMatch twice regardless, since I have to check it both ways.

Reply:
Would adding comments help instead? Just to explain what we're checking?

Reply:
I still think there are benefits to having all the matching logic in one function. It is easier to understand, can be unit tested, and will prevent future misuse (like someone removing the second wildcard check here).